src/tools/ToolSelectMove.ts

```diff
@@ -0,0 +1,147 @@
+          GlobalStore.commit("toggleSelectedDots", [existingDotIndex]);
+        }
+      }
+      else {
+        if (!event.shiftKey) {
+          GlobalStore.commit("clearSelectedDots");
+        }
+        if (event.altKey) {
+          GlobalStore.commit("toggleSelectedDots", [existingDotIndex]);
+        } else {
+          GlobalStore.commit("addSelectedDots", [existingDotIndex]);
+        }
+      }
+      this.moveToolStart = [x, y];
+      this.doGrapherToolMoveAction([x, y]);
+    } else {
+      // if we hvae not clicked on a dot, start a new selection.
+      if (!event.shiftKey) {
+        GlobalStore.commit("clearSelectedDots");
+      }
+      GlobalStore.commit("setSelectionLasso", [[x, y]]);
+      this.selectionLassoStart = [x, y];
+    }
+  }
+
+  onMouseUpInternal(event: MouseEvent): void {
+    const [x, y] = BaseTool.convertClientCoordinates(event);
+
+    if (this.moveToolStart !== null) {
+      // We are in a Move, so translate all the dots.
+      const [deltaX, deltaY] = [
+        x - this.moveToolStart[0],
+        y - this.moveToolStart[1],
+      ];
+      const currentSSDots: StuntSheetDot[] =
+        GlobalStore.getters.getSelectedStuntSheet.stuntSheetDots;
+      GlobalStore.state.selectedDots.forEach((index: number) => {
+        const [roundedX, roundedY] = BaseTool.roundCoordinateToGrid([
+          currentSSDots[index].x + deltaX,
+          currentSSDots[index].y + deltaY,
+        ]);
+        GlobalStore.commit("moveDot", {
+          index: index,
+          position: [roundedX, roundedY],
+        });
+      });
+      // Set the ToolDots to be empty to indicate we're not moving anymore.
+      GlobalStore.commit("setGrapherToolDots", []);
+      // null out moveToolStart to incidcate we're not moving anymore.
+      this.moveToolStart = null;
+      return;
+    }
+    if (this.selectionLassoStart !== null) {
+      // Complete the selection by finding everything in the selection box.
+      const stuntSheetDots: StuntSheetDot[] =
+        GlobalStore.getters.getSelectedStuntSheet.stuntSheetDots;
+      const newArray = stuntSheetDots.filter((dot) =>
+        InsideLasso(GlobalStore.state.selectionLasso, [dot.x, dot.y])
+      );
+      const newIndices = newArray.map((dot) =>
+        stuntSheetDots.findIndex((dot2): boolean => {
+          return dot.x === dot2.x && dot.y === dot2.y;
+        })
+      );
+      if (event.altKey) {
+        GlobalStore.commit("toggleSelectedDots", newIndices);
+      } else {
+        GlobalStore.commit("addSelectedDots", newIndices);
+      }
+      // we are done selecting, so clear out the box
+      GlobalStore.commit("setSelectionLasso", []);
+      this.selectionLassoStart = null;
+    }
+  }
+
+  onMouseMoveInternal(event: MouseEvent): void {
+    const [x, y] = BaseTool.convertClientCoordinates(event);
+    if (this.moveToolStart !== null) {
+      this.doGrapherToolMoveAction([x, y]);
+      return;
+    }
+    if (this.selectionLassoStart !== null) {
```

Comment:
nit: same here, can we do `else if`?

Reply:
I prefer early returns as I believe having code further on the left hand side with less indents are clearer.

Reply:
I prefer code closer to the left hand side.  So I like to avoid else ifs if it can be shown there is no other code to execute.